stylix/mk-target.nix

```diff
@@ -0,0 +1,73 @@
+{
+  # name used for creation of options
+  name,
+
+  # name used for description of options
+  humanName,
+
+  # condition for automatically enabling this target
+  autoEnable ? true,
+
+  # options to be added under `options.stylix.targets.${name}`
+  extraOptions ? { },
+
+  # list of functions with args representing Stylix options; function results
+  # only merged if all relevant Stylix options are not null
+  configElements ? [ ],
+
+  # function with args representing Stylix options; function result always
+  # merged
+  generalConfig ? null,
+}:
+{ config, lib, ... }:
+let
+  cfg = config.stylix.targets.${name};
+in
+{
+  imports = [
+    {
+      options.stylix.targets.${name}.enable =
+        config.lib.stylix.mkEnableTarget humanName autoEnable;
+    }
+    {
+      options.stylix.targets.${name} = extraOptions;
+    }
+  ];
+
+  config =
+    let
+      functionArgNames =
+        fn:
+        lib.pipe fn [
+          lib.functionArgs
+          builtins.attrNames
+        ];
+      getStylixAttrs =
+        fn:
+        lib.genAttrs (functionArgNames fn) (
+          arg:
+          if arg == "cfg" then
+            cfg
+          else if arg == "colors" then
+            config.lib.stylix.${arg}
+          else
+            config.stylix.${arg} or throw
+              "stylix: Stylix arg was requested by mk-target but does not exist: ${arg}"
```

Comment:
Believe it or not, this is effectively:

```nix
(config.stylix.${arg} or throw)
    "stylix: Stylix arg was requested by mk-target but does not exist: ${arg}"
```

Right now, so in the happy-path where `${arg}` is present, you will call it like a function, passing it the error string!

(nixfmt is even hinting at this odd [order-of-precedence](https://nix.dev/manual/nix/2.28/language/operators) by formatting the first and second part of the expression on separate lines)

The correct impl is to parenthesize the throw expression:

```nix
config.stylix.${arg} or
    (throw "stylix: Stylix arg was requested by mk-target but does not exist: ${arg}")
```